macro/src/test/scala/monocle/macros/MacroTests.scala

```diff
@@ -0,0 +1,32 @@
+package monocle.macros
+
+import probably._
+
+case class Street(number: Int, name: String)
+case class Address(city: String, street: Street)
+case class Company(name: String, address: Address)
+case class Employee(name: String, company: Company)
+
+sealed trait Json
+case class JNum(num: Double) extends Json
+
+object MacroTests extends Suite("Monocle Macro Tests") {
+  def run(test: Runner): Unit = {
+
+    test("construct Lens[Employee, Address] using GenLens") {
+      GenLens[Employee](_.company.address)
+    }.check(_ => true)
+
+    test("construct Lens[Employee, Int] using GenLens") {
+      GenLens[Employee](_.company.address.street.number)
+    }.check(_ => true)
```

Comment:
You could prefix this with `for(i <- 1 to 100)` to run the test a hundred times, if you wanted to see how consistent it is.